src/sleep/sleep.c

```diff
@@ -337,6 +347,12 @@ static int execute_s2h(const SleepConfig *sleep_config) {
 
                 log_debug("Battery discharge rate is %d%% per hour", estimated_discharge_rate);
 
+                if (estimated_discharge_rate > 0 && estimated_discharge_rate < 200) {
```

Comment:
this check exists twice now, please add a helper to unify this.

Reply:
Sure

Reply:
adding this check as part of put_battery_discharge_rate(), since we are adding the value only when it is in valid range.

Reply:
It appears in get_battery_discharge_rate also. Will add separate helper only then. Can be utilised there also.